jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpDateTime.java

```diff
@@ -0,0 +1,229 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Calendar;
+import java.util.Objects;
+import java.util.StringTokenizer;
+
+import org.eclipse.jetty.util.Index;
+import org.eclipse.jetty.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HTTP Date/Time parsing and formatting.
+ *
+ * <p>
+ *     Also covers RFC6265 Cookie Date parsing and formatting.
+ * </p>
+ */
+public class HttpDateTime
+{
+    private static final Logger LOG = LoggerFactory.getLogger(HttpDateTime.class);
+
+    private static final Index<Integer> MONTH_CACHE = new Index.Builder<Integer>()
+        .caseSensitive(false)
+        // Note: Calendar.Month fields are zero based.
+        .with("Jan", Calendar.JANUARY + 1)
+        .with("Feb", Calendar.FEBRUARY + 1)
+        .with("Mar", Calendar.MARCH + 1)
+        .with("Apr", Calendar.APRIL + 1)
+        .with("May", Calendar.MAY + 1)
+        .with("Jun", Calendar.JUNE + 1)
+        .with("Jul", Calendar.JULY + 1)
+        .with("Aug", Calendar.AUGUST + 1)
+        .with("Sep", Calendar.SEPTEMBER + 1)
+        .with("Oct", Calendar.OCTOBER + 1)
+        .with("Nov", Calendar.NOVEMBER + 1)
+        .with("Dec", Calendar.DECEMBER + 1)
+        .build();
+
+    private HttpDateTime()
+    {
+    }
+
+    /**
+     * Similar to {@link #parse(String)} but returns unix epoch
+     *
+     * @param datetime the Date/Time to parse.
+     * @return unix epoch in milliseconds, or -1 if unable to parse the input date/time
+     */
+    public static long parseToEpoch(String datetime)
+    {
+        try
+        {
+            Instant instant = parse(datetime);
+            return instant.toEpochMilli();
```

Comment:
Add a `ZonedDateTime parse(String)` method to avoid going via an `Instant` instance, as you can directly get the epoch from the `ZonedDateTime`, which is used in parse anyway.